### DISCUSSION TITLE
Bouton impulsionnel

### DISCUSSION THREAD

**Discussion by NicSamledernier** [witsa/synapps#558] [Q&A]
Bonjour,

Je n'arrive pas à créer un bouton impulsionnel pour commander une ressource...
J'ai besoin d'un bouton avec lequel un clic suffise à lancer une ressource booléenne.
Avec les synoptiques il faut cibler un bloc tempo qui agit sur le .InLink de la ressource logique à commander.
Avec Synapps qu'en est-il? Et quel acteur est approprié pour ce type de commande?
J'ai essayé le bouton qui cible un bloc tempo... c'est visiblement pas ça ou je manque quelque-chose.

**Comment by WITNicolas** [comment]
Bonjour,
Il vous faudrait une ressource de type CONSIGNE LOGIQUE.
Mettre un bloc TRIGGER du .setpoint vers le .set en mettant dans le TRIGGER le RISE à une valeur désirée en Secondes.
Automatiquement au clic sur la ressource, le trigger se déclenche et envoi par la suite vers le .set qui remet à 0 la ressource.

ça c'est pour la mécanique de base.

A partir de là dans STUDIO, soit vous utilisez l'acteur DETAIL DE REFLET, soit un acteur de type SWITCH INTERRUPTEUR et vous rélaisez votre liaison vers le .setpoint de la ressource.

**Comment by NicSamledernier** [comment]
Super merci!

**Comment by manufitoussiwit** [comment]
> Voir plus bas une meilleur version 🔽


Bonjour,

J'ai un composite qui pourrait vous aider : 

![image](https://user-images.githubusercontent.com/35595723/209192050-b62ba22c-bb70-48a3-863f-7bf7a756cc0e.png)

> C'est un bouton auquel on fournit l'adresse d'une variable booléenne sur laquelle on réalise une impulsion avec un délai en milliseconde réglable.

```text
SYNAPPS-STUDIO-COMPOSITE|{"config":{"key":"impulse","name":"Impulsion","additionalDefs":{"content":{"type":"text","label":"Contenu","value":"Impulsion"},"impulsePath":{"type":"wos-path","label":"Impulsion","value":"","relativeToPath":":easy.RESS"},"delay":{"type":"number","label":"Délai","value":700,"min":50,"step":50,"unit":"ms"}}},"leadActor":{"type":"layout/stack","key":"stack3","children":[{"type":"input/button","key":"button1","additionalDefs":{"impulsePath":{"type":"wos-path","label":"Impulsion","relativeToPath":":easy.RESS"},"delay":{"type":"number","label":"Délai","min":50,"step":50,"unit":"ms"}},"additionals":{"delay":150},"bindings":{"properties.enabled":"stage@properties.impulsePath","additionals.impulsePath":"stage@properties.impulsePath","additionals.delay":"stage@properties.delay","properties.content":"stage@properties.content"},"events":{"onClick":["context.getActor('variable').data = true;","setTimeout(() => {","  if (this.isDestroyed || this.isDestroying) return;","  context.getActor('variable').data = false;","}, this.additionals.delay);"]}},{"type":"redy/data-source/wos-variable","key":"ress","properties":{"autoRefresh":false,"mode":"relative","path":":easy.RESS"}},{"type":"redy/data-source/wos-relative-variable","key":"variable","properties":{"dataReadMode":"always","relativeTo":"ress","fieldName":"value","writeOnChange":true},"bindings":{"properties.relativePath":"actor#button1@additionals.impulsePath"}}]}}
```

A copier/coller dans les composites.

**Reply by WITNicolas** [reply]
Impecc le composite, ça peut être utile.

Je mets juste un petit bemol: 
l'impulsion ne fonctionnera qu'a partir de STUDIO et plus du tout en local dans la ressource du REDY.
Juste faire attention à cela lorsque l'on est en exploitation.

Dans le cas de ce composite, c'est STUDIO qui se charge de générer une impulsion.
Ce qui fait qu'on externalise un process au REDY qui logiquement est sensé garder le controle (au niveau process j'entends bien sûr :) )

**Reply by manufitoussiwit** [reply]
Bien vu!

Je modifie le composite pour que ce soit optionnel : 

![image](https://user-images.githubusercontent.com/35595723/210213212-e5562b5a-6479-4f80-9773-72930e45602b.png)

```
SYNAPPS-STUDIO-COMPOSITE|{"config":{"key":"impulse","name":"Impulsion","additionalDefs":{"content":{"type":"text","label":"Contenu","value":"Impulsion"},"impulsePath":{"type":"wos-path","label":"Impulsion","value":"","relativeToPath":":easy.RESS"},"withBack":{"type":"boolean","label":"Retour à 0?","value":false},"delay":{"type":"number","label":"Retour après","value":700,"min":50,"step":50,"unit":"ms"}}},"leadActor":{"type":"layout/stack","key":"stack3","children":[{"type":"input/button","key":"button1","additionalDefs":{"impulsePath":{"type":"wos-path","label":"Impulsion","relativeToPath":":easy.RESS"},"delay":{"type":"number","label":"Délai","min":50,"step":50,"unit":"ms"}},"additionals":{"delay":150},"bindings":{"properties.enabled":"stage@properties.impulsePath","additionals.impulsePath":"stage@properties.impulsePath","additionals.delay":"stage@properties.delay","properties.content":"stage@properties.content"},"events":{"onClick":["context.getActor('variable').data = true;","if (!this.stage.properties.withBack) return;","setTimeout(() => {","  if (this.isDestroyed || this.isDestroying) return;","  context.getActor('variable').data = false;","}, this.additionals.delay);"]}},{"type":"redy/data-source/wos-variable","key":"ress","properties":{"autoRefresh":false,"mode":"relative","path":":easy.RESS"}},{"type":"redy/data-source/wos-relative-variable","key":"variable","properties":{"dataReadMode":"always","relativeTo":"ress","fieldName":"value","writeOnChange":true},"bindings":{"properties.relativePath":"actor#button1@additionals.impulsePath"}}]}}
```

**Reply by Mathis554** [reply]
Bonjour à tout le monde;
Pour ce qui souhaite avoir un bouton avec juste une icône ou bien rajouter une icône avant le texte je vous, mais un composite ci-dessous :
`SYNAPPS-STUDIO-COMPOSITE|{"config":{"key":"impulse","name":"Bp","additionalDefs":{"contenta":{"type":"text","label":"Contenu","value":"Impulsion"},"impulsePath":{"type":"wos-path","label":"Impulsion","value":"","relativeToPath":":easy.RESS"},"withBack":{"type":"boolean","label":"Retour à 0?","value":true},"delay":{"type":"number","label":"Retour après","value":700,"min":50,"step":50,"unit":"ms"},"content":{"type":"icon","value":"buttona"},"prop1":{"type":"text","value":""}}},"leadActor":{"type":"layout/stack","key":"-","children":[{"type":"input/button","key":"button-icon1","properties":{"content":"<i class=\"icon-{{icon}}\"></i> {{text}}","backgroundColor":"rgba(255, 255, 255, 0)","borderColor":"rgba(255, 255, 255, 0)","borderStyle":"solid","boxShadow":"2px 2px 3px 0px rgba(255, 255, 255, 0)"},"additionalDefs":{"icon":{"type":"icon","label":"Icône","description":"Web Hosting Hub Glyphs"},"text":{"type":"text","label":"Texte"},"impulsePath":{"type":"wos-path","label":"Impulsion","relativeToPath":":easy.RESS"},"delay":{"type":"number","label":"Délai","min":50,"step":50,"unit":"ms"}},"additionals":{"icon":"ok","text":" ","delay":150},"bindings":{"additionals.icon":"stage@properties.content","properties.enabled":"stage@properties.impulsePath","additionals.impulsePath":"stage@properties.impulsePath","additionals.delay":"stage@properties.delay","additionals.text":"stage@properties.prop1"},"events":{"onClick":["context.getActor('variable').data = true;","if (!this.stage.properties.withBack) return;","setTimeout(() => {","  if (this.isDestroyed || this.isDestroying) return;","  context.getActor('variable').data = false;","}, this.additionals.delay);"]}},{"type":"redy/data-source/wos-variable","key":"ress","properties":{"autoRefresh":false,"mode":"relative","path":":easy.RESS"}},{"type":"redy/data-source/wos-relative-variable","key":"variable","properties":{"dataReadMode":"always","relativeTo":"ress","fieldName":"value","writeOnChange":true},"bindings":{"properties.relativePath":"actor#button-icon1@additionals.impulsePath"}}]}}`
